src/server/src/main/java/io/cassandrareaper/service/RepairScheduleService.java

```diff
@@ -117,6 +123,71 @@ public RepairSchedule storeNewRepairSchedule(
         .adaptive(adaptive)
         .percentUnrepairedThreshold(percentUnrepairedThreshold);
 
-    return context.storage.addRepairSchedule(scheduleBuilder);
+    RepairSchedule repairSchedule = context.storage.addRepairSchedule(scheduleBuilder);
+    registerScheduleMetrics(repairSchedule.getId());
+    return repairSchedule;
+  }
+
+  public void deleteRepairSchedule(UUID repairScheduleId) {
+    unregisterScheduleMetrics(repairScheduleId);
+    context.storage.deleteRepairSchedule(repairScheduleId);
+  }
+
+  private void registerRepairScheduleMetrics(Collection<RepairSchedule> allRepairSchedules) {
+    allRepairSchedules.forEach(schedule -> registerScheduleMetrics(schedule.getId()));
+  }
+
+
+  private void registerScheduleMetrics(UUID repairScheduleId) {
+    RepairSchedule schedule = context.storage.getRepairSchedule(repairScheduleId).get();
+    RepairUnit repairUnit = context.storage.getRepairUnit(schedule.getRepairUnitId());
+    String metricName = metricName("millisSinceLastRepairForSchedule",
+            repairUnit.getClusterName(),
+            repairUnit.getKeyspaceName(),
+            schedule.getId());
+
+    if (!context.metricRegistry.getMetrics().containsKey(metricName)) {
+      context.metricRegistry.register(metricName, getMillisSinceLastRepairForSchedule(schedule.getId()));
+    }
+  }
+
+  private void unregisterScheduleMetrics(UUID repairScheduleId) {
+    RepairSchedule schedule = context.storage.getRepairSchedule(repairScheduleId).get();
+    RepairUnit repairUnit = context.storage.getRepairUnit(schedule.getRepairUnitId());
+    String metricName = metricName("millisSinceLastRepairForSchedule",
+        repairUnit.getClusterName(),
+        repairUnit.getKeyspaceName(),
+        schedule.getId());
+
+    if (context.metricRegistry.getMetrics().containsKey(metricName)) {
+      context.metricRegistry.remove(metricName);
+    }
+  }
+
+  private Gauge<Long> getMillisSinceLastRepairForSchedule(UUID repairSchedule) {
+    return () -> {
+      Optional<RepairSchedule> schedule = context.storage.getRepairSchedule(repairSchedule);
+
+      Long lastRepairDate = schedule.orElseThrow(() -> new IllegalArgumentException("Repair schedule not found"))
+          .getRunHistory().stream()
```

Comment:
`runHistory` is no longer updated in the repair schedule table because it was unused and could create very large collections over time.
You can use `lastRun` instead which gives the id of the last repair run for this schedule.

Reply:
Is there a reason why the runHistory is still present in the code if it is no longer kept up to date?

Looking at the code, it looks like the lastRun is updated immediately upon creating a new repair run, before it has completed. I suspect this will cause the metric to get assigned the default value when a new repair for the schedule starts and until the repair is completed. 

I'll write a couple of tests to confirm this (as mentioned in the other comments).

Reply:
> Is there a reason why the runHistory is still present in the code if it is no longer kept up to date?

Not that I know of, we should've removed it.
We should then change the code so that `lastRun` gets updated at the end of the repair instead of at its beginning. This way it won't mess the metrics.

Reply:
I have rewritten the code to use lastRun and changed when lastRun is updated.

Reply:
Awesome! I'll review the PR today.